src/plugins/es_ui_shared/__packages_do_not_import__/errors/es_error_parser.ts

```diff
@@ -0,0 +1,52 @@
+/*
```

Comment:
I just wanted to flag that this functionality is similar to what is in `x-pack/plugins/ingest_pipelines/public/application/components/pipeline_form/pipeline_form_error/error_utils.ts`, but that code has an added, perhaps unnecessary, runtime type guard since the assumption was there are no guarantees on what the error object will be (`unknown`).

I see this is used on the server side and we guard with the "isEsError" check.

We can replace `flattenErrorsTree` in ingest pipelines with this logic to get maximum re-use because this should also work for the errors returned for processors if this is the standard es error shape. I think we still need to guard in ingest pipelines with the runtime type check though.

Let me know what you think!

Reply:
Yes, indeed it is similar but much simpler implementation :) If you think your use case deserves all the complexity of `io-ts` and `fp-ts` then it's better to have separate implementations. My goal was "just" to read the "caused_by" recursively from the ES error. I put the logic inside a `try/catch` so it should not break if ES changes the contract.

I personally think that sometimes too much typings can get in the way and simplicity is often a better approach. That's just my personal preference though. 😊